Runtimes/Core/cmake/modules/CatalystSupport.cmake

```diff
@@ -0,0 +1,26 @@
+# Add flags for generating the zippered target variant in the build
+
+# Initialize `SwiftCore_VARIANT_MODULE_TRIPLE` if the driver is able to emit
+# modules for the target variant.
+
+if(SwiftCore_TARGET_VARIANT_COMPILER_TARGET)
+  add_compile_options(
+    "$<$<COMPILE_LANGUAGE:C,CXX>:SHELL:-darwin-target-variant ${SwiftCore_TARGET_VARIANT_COMPILER_TARGET}>"
+    "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-target-variant ${SwiftCore_TARGET_VARIANT_COMPILER_TARGET}>")
```

Comment:
After further digging, `-target-variant` doesn't seem to fully work when linking with the swift driver.
The resulting dylib is not zippered, even though the objects are. Continuing to investigate.